### PR TITLE
fix(pie): use legend.data property if supplied

### LINKED ISSUE #1439: Legends in Pie chart ignore data property
I found this problem while trying to setup legends on multiple lines (as in [this issue](https://github.com/plouc/nivo/issues/433))

A practical example can be found [here](https://codesandbox.io/s/elated-pike-lp7h8).

Line chart will render legends correctly with the passed custom data.
On the other hand, Pie will render legends with data passed to the chart, not to legends.

Is there a problem with my implementation? Is it not intended for Pie chart to render legends with custom data?

### PR DESCRIPTION
**Pull request by wyze**
Close #1439

### CODE REVIEW

File: packages/pie/src/PieLegends.tsx [hunk 1]
```diff
@@ -23,7 +23,7 @@ const PieLegends = <RawDatum,>({
                     {...legend}
                     containerWidth={width}
                     containerHeight={height}
-                    data={dataWithArc}
+                    data={legend.data ?? dataWithArc}
```
Comment (plouc) [inline packages/pie/src/PieLegends.tsx:26]: As [we already have some unit tests for the legends](https://github.com/plouc/nivo/blob/master/packages/pie/tests/Pie.test.tsx#L518), I think we could add a test case for this, what do you think?

**Review by plouc — CHANGES_REQUESTED** [review]

File: packages/pie/src/PieLegends.tsx (hunk 1 above)
Comment (wyze) [reply packages/pie/src/PieLegends.tsx:26]: Oh, yes definitely will add a test. Good catch.

Comment (plouc) [reply packages/pie/src/PieLegends.tsx:26]: Thanks!

**Review by plouc — APPROVED** [review]
LGTM

**Pull request merged**